lesson4/argirovatina_hometask.rb

```diff
@@ -0,0 +1,40 @@
+class LazyHash < Hash
+
+  def method_missing(method_name)
+    method_name = get_correct_key(method_name)
```

Comment:
oh, I found another problem with it. if there are no such `hash_key` as a `String` or a `Symbol` your method returns `nil`. this `nil` will be used as `method_name` in which is not right.

but if you change it like
```ruby
key = get_correct_key(method_name)
if key
  self[key] = self[key].call
else
  raise "Method #{method_name} is missing"
end
```
In that case, we leave `method_name` 100% unchanged. It's ok for us, we don't care about type of this object, anyway interpolation would convert it into string in `raise "Method #{method_name} is missing"` line.